src/vs/workbench/contrib/terminal/browser/terminalInstance.ts

```diff
@@ -1126,10 +1126,10 @@ export class TerminalInstance extends Disposable implements ITerminalInstance {
 		return this.xterm ? this.xterm.raw.hasSelection() : false;
 	}
 
-	async copySelection(): Promise<void> {
+	async copySelection(asHtml?: boolean): Promise<void> {
 		const xterm = await this._xtermReadyPromise;
 		if (this.hasSelection()) {
-			await this._clipboardService.writeText(xterm.raw.getSelection());
+			asHtml ? await xterm.copyAsHtml() : await this._clipboardService.writeText(xterm.raw.getSelection());
```

Comment:
Instead of `XtermTerminal.copyAsHtml` we should only get the selection so we can use the clipboard service to do the copying. Something more like:

```ts
await this._clipboardService.writeText(asHtml ? xterm.getSelectionAsHtml() : xterm.raw.getSelection());
```

Reply:
That doesn't work because we need to set the type to `text/html` and the clipboard service only handles `text`. See the exploration above

Reply:
it does use the selection https://github.com/microsoft/vscode/pull/144784/files#diff-7556fb25bb884748b962216539e5915930478d5adcf42860a861d70e49c36b9cR179

Reply:
@meganrogge oh I see, I figured the editor would be using the clipboard service but it's just using `document.execCommand('copy')`:

https://github.com/microsoft/vscode/blob/35f8ba6f6678a818f8d3e994643bd9f167c9c422/src/vs/editor/contrib/clipboard/browser/clipboard.ts#L178-L181

In that case can we still move the clipboard handling into TerminalInstance so all dealing with the clipboard is at the same level?